src/main/java/com/artipie/RepoConfig.java

```diff
@@ -128,17 +128,32 @@ public URL url() {
      * @return Async storage for repo
      */
     public Storage storage() {
-        final YamlMapping repo = this.repoConfig();
-        final Storage storage;
-        final YamlNode node = repo.value("storage");
-        if (node instanceof Scalar) {
-            storage = this.storages.storage(((Scalar) node).value());
-        } else if (node instanceof YamlMapping) {
-            storage = new YamlStorage((YamlMapping) node).storage();
-        } else {
-            throw new IllegalStateException(String.format("Invalid storage config: %s", node));
-        }
-        return new SubStorage(this.prefix, new LoggingStorage(Level.INFO, storage));
+        return this.storageOpt().orElseThrow(
+            () -> new IllegalStateException("Storage is not configured")
+        );
+    }
+
+    /**
+     * Create storage if configured.
+     *
+     * @return Async storage for repo
+     */
+    public Optional<Storage> storageOpt() {
```

Comment:
@olegmoz I think we can replace `storage()` with this implementation and just call `storage().orElseThrow()` to get storage where it's required

Reply:
@g4s8 I think I did just that, `RepoConfig.storage()` method looks like that now:
```
public Storage storage() {
        return this.storageOpt().orElseThrow(
            () -> new IllegalStateException("Storage is not configured")
        );
    }
```